src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java

```diff
@@ -163,9 +167,9 @@ public void generateStructure(StructureCallback callback) {
             minNoisyR2 *= minNoisyR2;
             // iterate through blocks
             Vector4f pos = new Vector4f();
-            for (int x = Math.max(0, minPosition.x()); x <= Math.min(chunkSize.x() - 1, maxPosition.x()); x++) {
-                for (int y = Math.max(0, minPosition.y()); y <= Math.min(chunkSize.y() - 1, maxPosition.y()); y++) {
-                    for (int z = Math.max(0, minPosition.z()); z <= Math.min(chunkSize.z() - 1, maxPosition.z()); z++) {
+            for (int x =  minPosition.x(); x <=  maxPosition.x(); x++) {
```

Comment:
It's fine to have the coordinates exceeding the bounds of the chunk, but if they exceed the bounds of the facet's borders, it will cause an exception.

Reply:
It applies an inverse matrix to map them from global to local space. we need to clip it at the end right?